### ISSUE TITLE
Truthy rule gives confusing result for common Github Actions yml that uses "on" syntax

### ISSUE THREAD

**Issue by AndrewLaneAtPowerSchool** [adrienverge/yamllint#540]
The truthy rule seems to see the "on" keyword in the following YAML and think it should be true/false (assuming because on/off can be seen as truthy and falsy?).

Example YML:
```
name: Run tests

on: [push, pull_request]

jobs:  
  test:
    runs-on: ubuntu-latest

    steps:
    - name: …
```

Snippet of yamllint output:

```
  3:1       warning  truthy value should be one of [false, true]  (truthy)
``

Expected: should not throw warning

**Comment by andrewimeson** [comment]
This is not a false positive. This is a documented and expected behavior. Your options are:

1. Quote the key

   ```yaml
   ---
   name: Run tests

   "on": [push, pull_request]
   ```

2. Disable the rule with a line comment

   ```yaml
   ---
   name: Run tests

   on: [push, pull_request]  # yamllint disable-line rule:truthy
   ```

3. Disable key checking in the yamllint configuration (e.g. `.yamllint.yaml`)

   ```yaml
   ---
   extends: default
   rules:
     truthy:
       check-keys: false
   ```


This is a duplicate of #430 and others

**Comment by AndrewLaneAtPowerSchool** [comment]
Thanks!
